src/Runner.hs

```diff
@@ -115,20 +151,28 @@ overwrite msg = do
   liftIO (hPutStr stderr str)
 
 -- | Run all examples from given module.
-runModule :: Bool -> Bool -> Interpreter -> Module [Located DocTest] -> Report ()
-runModule fastMode preserveIt repl (Module module_ setup examples) = do
+runModule
+  :: Bool
+  -> Bool
+  -> Either [String] Interpreter
+  -> Chan ReportUpdate
+  -> Module [Located DocTest]
+  -> IO ()
+runModule fastMode preserveIt (Left args) output mod_ = do
+  Interpreter.withInterpreter args $ \repl -> withCP65001 $
+    runModule fastMode preserveIt (Right repl) output mod_
+
+runModule fastMode preserveIt (Right repl) output (Module module_ setup examples) = do
+  successes <- mapM (runTestGroup preserveIt repl reload output) setup
 
-  Summary _ _ e0 f0 <- gets reportStateSummary
+  -- only run tests, if setup does not produce any errors/failures
+  when (and successes) (mapM_ (runTestGroup preserveIt repl setup_ output) examples)
```

Comment:
It'd be nice to have a test for this behaviour.

Reply:
Sure thing! There might already be a test for this, as this isn't changed behavior. I'll take a look.